--- PixivBase.php.orig	2019-09-07 15:02:50.126299645 +0900
+++ PixivBase.php	2019-09-07 15:10:46.990992022 +0900
@@ -30,17 +30,17 @@
     /**
      * @var string
      */
-    protected $oauth_client_id = 'bYGKuGVw91e0NMfPGp44euvGt59s';
+    protected $oauth_client_id = 'MOBrBDS8blbauoSck0ZfDbtuzpyT';
 
     /**
      * @var string
      */
-    protected $oauth_client_secret = 'HP3RmkgAmEGro0gn1x9ioawQE8WMfvLXDz3ZqxpK';
+    protected $oauth_client_secret = 'lsACyCD94FhDUtGTXi3QzcFE2uU1hqtDaKeqrdwj';
 
     /**
      * @var string
      */
-    protected $oauth_device_token = 'af014441a5f1a3340952922adeba1c36';
+    protected $oauth_hash_secret = '28c1fdd170a5204386cb1313c7077b34f83e4aaf4aa829ce78c231e05b0bae2c';
 
     /**
      * @var null
@@ -76,7 +76,7 @@
         $request = array(
             'client_id' => $this->oauth_client_id,
             'client_secret' => $this->oauth_client_secret,
-            'device_token' => $this->oauth_device_token,
+            'get_secure_url' => '1',
         );
         if ($user != null && $pwd != null) {
             $request = array_merge($request, array(
@@ -96,7 +96,12 @@
         $curl->setOpt(CURLOPT_CONNECTTIMEOUT, 10);
         $curl->setOpt(CURLOPT_SSL_VERIFYHOST, 0);
         $curl->setOpt(CURLOPT_SSL_VERIFYPEER, 0);
-        $curl->setHeader('Authorization', $this->headers['Authorization']);
+
+        $localTime = (new \DateTime())->format(DateTime::ATOM);
+        $clientHash = md5($localTime . $this->oauth_hash_secret);
+        $curl->setHeader('X-Client-Time', $localTime);
+        $curl->setHeader('X-Client-Hash', $clientHash);
+
         $curl->post($this->oauth_url, $request);
         $result = $curl->response;
         $curl->close();
